src/main.rs

```diff
@@ -391,6 +401,13 @@ impl SpiDrv {
             Err(e) => { return Err(e); }
         }
     }
+
```

Comment:
I have had a hard time understanding this function and its purpose. I wonder if adding some comments would make it clearer. I know we copied it from the cpp code. Documenting this is certainly not a blocker.